packages/block-editor/src/store/actions.js

```diff
@@ -394,16 +394,18 @@ export const replaceBlocks =
 				return;
 			}
 		}
-		dispatch( {
-			type: 'REPLACE_BLOCKS',
-			clientIds,
-			blocks,
-			time: Date.now(),
-			indexToSelect,
-			initialPosition,
-			meta,
+		registry.batch( () => {
+			dispatch( {
+				type: 'REPLACE_BLOCKS',
+				clientIds,
+				blocks,
+				time: Date.now(),
+				indexToSelect,
+				initialPosition,
+				meta,
+			} );
+			dispatch.ensureDefaultBlock();
```

Comment:
Suggestion (non-blocking): Should we also add a comment similar to `privateRemoveBlocks` here? It makes it easier to remember the context of why the action is dispatched.

Reply:
Comment that explains that the batching is there because of undo would also be nice, it's not obvious.